src/capabilities/capabilities.ts

```diff
@@ -49,11 +54,52 @@ export function fetchCapabilitiesOnFocus(): Promise<void> {
 }
 
 function resolveCapabilities(resolve: () => void): void {
-  void getCapabilities().then((capabilities) => {
-    capabilities?.enabled_features.forEach((feature) => {
-      enabledCapabilities[feature] = true;
-    });
-
+  void refreshCapabilities().then(() => {
     resolve();
+    startRefreshLoop();
+  });
+}
+
+const capabilitiesRefreshed = new EventEmitter<void>();
+
+export function onDidRefreshCapabilities(listener: () => void): Disposable {
+  return capabilitiesRefreshed.event(listener);
+}
+
+async function refreshCapabilities(): Promise<void> {
+  const capabilities = await getCapabilities();
+
+  enabledCapabilities = {};
+  capabilities?.enabled_features.forEach((feature) => {
```

Comment:
```suggestion
  enabledCapabilities = capabilities?.enabled_features.map((feature) => ({[feature]: true})
```

Reply:
This does not do the same thing

```
> let enabled_features = ['a', 'b', 'c'];
undefined
> 
> let enabledCapabilities = {};
undefined
> enabled_features.forEach((feature) => {
...     enabledCapabilities[feature] = true;
... });
undefined
> 
> enabledCapabilities
{ a: true, b: true, c: true }
> 
> enabled_features.map((feature) => ({[feature]: true}))
[ { a: true }, { b: true }, { c: true } ]
```

To use it properly, we can do something like

```
Object.assign({}, ...enabled_features.map((feature) => ({[feature]: true})))
```

Or maybe use `Array.reduce`

```
enabled_features.reduce((o, feature) => {
	o[feature] = true;
	return o;
}, {});
```

Both do not seem more readable to me than the original